File: backend/app/core/entities/order.py
from pydantic import BaseModel
from datetime import datetime
from typing import List

class OrderItem(BaseModel):
    name: str
    quantity: int
    price_per_unit: int = 0
    total: int = 0

class OrderRound(BaseModel):
    created: datetime
    items: List[OrderItem]


class Order(BaseModel):
    created: datetime
    paid: bool = False
    subtotal: int = 0
    taxes: int = 0
    discount: int = 0
    items: List[OrderItem]
    rounds: List[OrderRound]

File: frontend/src/app/types/Order.ts
export interface OrderItem {
    name: string;
    quantity: number;
    price_per_unit: number;
    total: number;
}

export interface Order {
    created: string;
    subtotal: number;
    total: number;
    taxes: number;
    discounts: number;
    items: OrderItem[];
}